docs/decisions/0018-expose-all-sass-vars-as-css-vars.rst

```diff
@@ -0,0 +1,143 @@
+  Paragon defaults and add additional SCSS code
+- Then we have MFEs that import from the branding package and Paragon, and
+  usually hav some of their own SCSS code
+
+Paragon uses Bootstrap variables, and the branding theme uses Paragon/Bootstrap
+variables, any changes to any variables mean that the entire codebase needs to
+be rebuilt. However, this need not be the case, if the dependencies between
+these packages is expressed via CSS variables instead of SCSS variables, then
+any changes to values will be picked up at runtime. Such a setup would allow
+building Bootstrap + Paragon + the branding package into a stylesheet that
+contains all the variables MFEs need so they can use the stylesheet directly
+and not need recompilation if this common stylesheet changes.
+
+This ADR only concerns itself consumers of Paragon, and as such still expects
+Paragon to continue using SCSS variables itself.
+
+If we want to be able to support a common pre-built core stylesheet that can be
+used by all MFEs and deployed independently of MFEs, then MFEs cannot use any
+SCSS variables from Paragon + Bootstrap since in that case any changes to the
+core theme will require a rebuild of each MFE, and as such these variables
+should instead be used as CSS variables so that they can be looked up at
+runtime rather then build time.
+
+Bootstrap already exports a number of CSS variables (these can be found in
+`this document for Bootstrap 4.x`_), however they don't cover all usecases. For
+instance, usage of colour variables like `primary-400`, or `light-300` are
+abundant in MFEs, however these are not expored by Bootstrap even though they
+are available as SCSS variables.
+
+Decision
+********
+
+We will export all theme variables that an MFE could need as CSS variables so
+that these can be applyed at runtime rather than build time.
+
+Paragon can create CSS variable exports like: `--pgn-blue`, `--pgn-primary`,
+`--pgn-primary-200`` etc. The reason for prefixing with `--pgn` is to isolate
+changes to these names in Bootstrap. For example, in Bootstrap 4.x these are
+named as `--blue, --primary` etc. However in `Bootstrap 5`_, they are prefixed
+with `--bs-blue, --bs-primary`, this incompatibility can be handled at the
+Paragon level whenever it switches from 4.x to 5.x without needing changes from
+MFEs.
+
+Paragon will pick up a standardised set of SCSS variables such as those
+relating to colours, colour shades, font styles, and responsible breakpoints
+and expose these as CSS variables with a ``--pgn`` prefix. These variabels will
+be exported by the Paragon package, however, they will be use variables defined
+in the branding package thus allowing them to be redefined easily.
+
+Any current usage of SCSS variables in MFEs can then be replaced with the
+equivalent CSS variable.
+
+
+Consequences
+************
+
+Once this is implemented, MFEs, and other consumers of Pargaon can start using
+CSS variables, wherever they are currently using SASS variables. The
```

Comment:
I think it'd be great to strive to theming capabilities with either CSS variables _or_ SCSS variables depending on the needs of the consumer/application. Is the intention here to no longer rely on SCSS variables all, and if so, it's worth calling out the impacts of that on consumers in terms of necessary refactors, etc. We'd want to ensure we minimize breaking changes as much as possible.